### ISSUE TITLE
Expose ADC_NUM_AVERAGING to configurator?

### ISSUE THREAD

**Issue by Skerper** [SimuCUBE/SimuCUBE-OpenSource-Firmware#3]
Mika,
Is there any chance you could switch this from being a define to being something adjusted by the configurator?

https://github.com/SimuCUBE/SimuCUBE-OpenSource-Firmware/blob/4df9b9d6a8089447bb74ef0920496eca7b07ecb9/Inc/cAnalogConfig.h#L29

This would allow configuring around spiky/jittery pots.

**Comment by mtakala** [comment]
We have actually changed this from
Sort the x latest samples, then discard largest and smallest values and average the rest
To
Low pass filter all the average of last x values, with around 100 Hz cutoff frequency. 

This has been implemented in Simucube 2, and will be published here as part of the next Simucube firmware open source release. 

What is the rationale to increase the number of samples?

This seems to be sufficient amd

**Comment by Skerper** [comment]
Oh, disregard this then - that sounds like a far better implementation. 

Will the Hz value of the low pass filter be configurable?

**Comment by mtakala** [comment]
No, its not configurable at the moment. I think spiky pedal potentiometers and such should be replaced anyway, instead of adding confusing software solutions.

**Comment by Skerper** [comment]
These are just cheapo trim pots for brake bias/bars/TC etc, so I was hopeful I could just set a big smoothing factor and not worry :D

**Comment by mtakala** [comment]
Ah, I thought everyone would be using incremental encoders for those kind of adjustments, support for them was implemented for 0.11.0 version and up and thus is not yet in this released version of open source code.

I'm leaving the issue open so I can remember to ask for testing result of those potentiometers when we get the next opensource release out.

**Comment by mtakala** [comment]
Next opensource release is  now out, and matches Simucube 1 v 1.0.24 code. How are the potentiometers working for you?

**Comment by Skerper** [comment]
Sorry for the very late reply. I have updated to latest, but I don't see any changes in the potentiometer behavior.

**Comment by mtakala** [comment]
They must be very, very spiky cheap pots then. Have you taken a look with an oscilloscope?

**Comment by Skerper** [comment]
They are indeed very cheap and very spiky. I was hoping it was all I needed, since these adjustments could have 3+ seconds of smoothing and still not have any repurcussions. Unfortunately I don't have access to an oscilloscope.
